cmd/cluster-network-renderer/main.go

```diff
@@ -69,7 +70,7 @@ func render() error {
 
 // readConfigObject reads a NetworkConfig object from disk
 func readConfigObject(path string) (*operv1.Network, error) {
-	f, err := os.Open(path)
+	f, err := os.Open(filepath.Clean(path))
```

Comment:
What is the gosec warning? Who says that this is good practice?

Reply:
The warning is:

[/home/josorior/go/src/github.com/openshift/cluster-network-operator/cmd/cluster-network-renderer/main.go:72] - G304: Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)

And the idea behind the warning is that, if a user has provided the path, they could potentially traverse the directory structure in unintended ways by inserting "../" in the given path. So, any path that's opened in the codebase that might come from a user (or an untrusted source) should be cleaned up: https://golang.org/pkg/path/filepath/#Clean

If you think this is not a problem in this codebase, we could easily just ignore this warning when invoking gosec.

Reply:
For sure - adding this line would *create* a bug. This is a CLI tool, so I would 100% expect the user to be able to traverse any path they want.

Reply:
That warning seems pretty specific to the case of code that responds to HTTP requests by combining a trusted path prefix with a user-provided suffix, and doesn't really apply to anything that we're doing in OpenShift. Additionally, the fact that adding `filepath.Clean()` like this makes the warning go away means that the analyzer isn't even doing a good job, because that doesn't actually fix the HTTP-server-combining-path security hole. (You'd need to clean the user-provided suffix separately *before* combining it with the trusted prefix.)

So I think we should be disabling this warning in *all* OpenShift codebases, because it's mostly not relevant, and it seems to encourage cargo-cult fixes that don't actually address the problem.

Reply:
alright, disabled this check.